### PR TITLE
i18n production support

### PR DESCRIPTION
**Pull request by domcleal**
.mo files added and locale domain name fixed so i18n works in prod, also extracted the latest strings + pulled from tx.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,7 +15,6 @@ tmp
 Gemfile.lock
 locale/*.mo
 locale/*/*.pox
-locale/*/LC_MESSAGES
```
Comment (lzap) [inline .gitignore:18]: Why is that? I think these subdirs are being created during the build...

File: locale/en_GB/foreman_discovery.po [hunk 2]
```diff
@@ -7,11 +7,11 @@
...
 msgstr ""
-"Project-Id-Version: Foreman\n"
+"Project-Id-Version: foreman_discovery 1.3.0\n"
 "Report-Msgid-Bugs-To: \n"
-"POT-Creation-Date: 2014-04-08 15:59+0200\n"
-"PO-Revision-Date: 2014-08-04 08:40+0000\n"
-"Last-Translator: Dominic Cleal <dcleal@redhat.com>\n"
+"POT-Creation-Date: 2014-08-07 10:19+0100\n"
+"PO-Revision-Date: 2014-08-08 09:41+0000\n"
+"Last-Translator: Lukáš Zapletal\n"
```
Comment (lzap) [inline locale/en_GB/foreman_discovery.po:14]: The born English expert. _cough_ _cough_ _cough_

File: lib/foreman_discovery/engine.rb [hunk 3]
```diff
@@ -18,7 +18,7 @@ class Engine < ::Rails::Engine
 
     initializer 'foreman_discovery.register_gettext', :after => :load_config_initializers do |app|
       locale_dir = File.join(File.expand_path('../../..', __FILE__), 'locale')
-      locale_domain = 'discovery'
+      locale_domain = 'foreman_discovery'
```
Comment (lzap) [inline lib/foreman_discovery/engine.rb:21]: Hmmm so this was not working at all...

**Comment by lzap** [comment]
:+1:

File: .gitignore (hunk 1 above)
Comment (domcleal) [inline .gitignore:18]: There is no build?  They need to be part of the gem, same as we did in Hammer.

It's only in core that we don't ship the mo files and do it in packaging.  For plugin installs, gems really need to contain them as they're installed directly.

File: locale/en_GB/foreman_discovery.po (hunk 2 above)
Comment (domcleal) [inline locale/en_GB/foreman_discovery.po:14]: Think Transifex attributes any .pot update to you...

File: lib/foreman_discovery/engine.rb (hunk 3 above)
Comment (domcleal) [inline lib/foreman_discovery/engine.rb:21]: Indeed!

File: .gitignore (hunk 1 above)
Comment (lzap) [inline .gitignore:18]: Isn't there a hook avaiable during the `gem build` so we can build it there? I'd rather prefer that and make it a standard.

Comment (lzap) [inline .gitignore:18]: Does not look like...

http://guides.rubygems.org/specification-reference/#post_install_message

Ok :+1:

Comment (domcleal) [inline .gitignore:18]: indeed... :(  only way you can hack around it IIRC is to pretend you have a native extension, it's not good

**Comment by lzap** [comment]
Merged, thanks:
- c267331 i18n - pulling from tx <Dominic Cleal>
- 0bc30a2 i18n - add Makefile to generate .mo files, fix domain name <Dominic Cleal>

**Pull request merged**